python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -2363,3 +2363,144 @@ def test_array_from_pylist_offset_overflow():
     assert isinstance(arr, pa.ChunkedArray)
     assert len(arr) == 2**31
     assert len(arr.chunks) > 1
+
+
+@parametrize_with_collections_types
+@pytest.mark.parametrize(('data', 'scalar_data', 'value_type'), [
+    ([True, False, None], [pa.scalar(True), pa.scalar(False), None], pa.bool_()),
+    ([1, 2, None], [pa.scalar(1), pa.scalar(2), None], pa.int64()),
```

Comment:
```suggestion
    ([1, 2, None], [pa.scalar(1), pa.scalar(2), pa.scalar(None, pa.int64())], pa.int64()),
```

To have a case with a null scalar instead of None as well